src/Surfnet/StepupMiddleware/CommandHandlingBundle/Pipeline/TransactionAwarePipeline.php

```diff
@@ -30,31 +32,81 @@ class TransactionAwarePipeline implements Pipeline
     /**
      * @var Connection
      */
-    private $connection;
+    private $middlewareConnection;
+    /**
+     * @var Connection
+     */
+    private $gatewayConnection;
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
 
     /**
-     * @param Pipeline $innerPipeline
-     * @param Connection $connection
+     * @param LoggerInterface $logger
+     * @param Pipeline        $innerPipeline
+     * @param Connection      $middlewareConnection
+     * @param Connection      $gatewayConnection
      */
-    public function __construct(Pipeline $innerPipeline, Connection $connection)
-    {
-        $this->innerPipeline = $innerPipeline;
-        $this->connection = $connection;
+    public function __construct(
+        LoggerInterface $logger,
+        Pipeline $innerPipeline,
+        Connection $middlewareConnection,
+        Connection $gatewayConnection
+    ) {
+        $this->logger               = $logger;
+        $this->innerPipeline        = $innerPipeline;
+        $this->middlewareConnection = $middlewareConnection;
+        $this->gatewayConnection    = $gatewayConnection;
     }
 
-    public function process($command)
+    public function process(Command $command)
     {
-        $this->connection->beginTransaction();
+        $this->logger->debug(sprintf(
+            'Starting Transaction in TransactionAwarePipeline for processing command "%s"',
+            $command
+        ));
+
+        $this->middlewareConnection->beginTransaction();
+        $this->gatewayConnection->beginTransaction();
```

Comment:
Thought about it, but adds inter-bundle dependency. Had a note on a post-it to discuss it actually as I'm not too fond about the dep, but frankly not too concerned about it either :)